torch/distributed/pipelining/_backward.py

```diff
@@ -320,6 +320,8 @@ def extract_tensors_with_grads(output_val, grad_val):
         torch.autograd.backward(
             stage_output_tensors, grad_tensors=output_grad_tensors  # type: ignore[arg-type]
         )
+        del stage_output_tensors
+        del output_grad_tensors
```

Comment:
Interesting, is this related to how `stage_backward` is called? i.e. lambda function
https://github.com/pytorch/pytorch/blob/3be150653cedb51f4588b15a46dc0522de672f9f/torch/distributed/pipelining/stage.py#L490-L495

Reply:
no, this is unrelated.
I'm adding additional information to the doc and PR-desc